File: lib/src/main/java/com/sha/formvalidator/validator/SuffixValidator.java
package com.sha.formvalidator.validator;

import android.widget.EditText;

/**
 * A validator that returns true only if the input field contains only numbers.
 *
 */
public class SuffixValidator extends Validator {

    private String suffix;

    public SuffixValidator(String suffix, String errorMessage) {
        super(errorMessage);
        this.suffix = suffix;
    }

    public boolean isValid(EditText et) {
        return et.getText().toString().endsWith(suffix);
    }
}

File: lib/src/main/java/com/sha/formvalidator/textview/validator/SuffixValidator.kt
package com.sha.formvalidator.textview.validator

import android.widget.TextView

/**
 * A validator that returns true only if the input field contains only numbers.
 *
 */
class SuffixValidator(private val suffix: String, errorMessage: String) : TextViewValidator(errorMessage) {

    override fun isValid(tv: TextView): Boolean = tv.text.toString().endsWith(suffix)
}
